api/src/main/java/net/kyori/adventure/text/renderer/ComponentFlattener.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2021 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.renderer;
+
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.util.Buildable;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * A 'flattener' to convert a component tree to a linear string for display.
+ *
+ * @since 4.7.0
+ */
+public interface ComponentFlattener extends Buildable<ComponentFlattener, ComponentFlattener.Builder> {
+  /**
+   * Create a new builder for a flattener.
+   *
+   * @return a new builder
+   * @since 4.7.0
+   */
+  static @NonNull Builder builder() {
+    return new ComponentFlattenerImpl.Builder();
+  }
+
+  /**
+   * A basic flattener that will only print information directly contained in components.
```

Comment:
```suggestion
   * A basic flattener that will print only information directly contained in components.
```

I know this is abstract API, but this statement made me start questioning what "information directly contained in components" means in general. I looked at the implementation to know what the intended meaning is here.

What seems intended here, to me, is that it specifically "does not print information not contained in the component" (which is fair; such as not doing translations). In my opinion, the original wording seems to go just short of implying 'all' types of information will be printed instead of letting the implementing flattener decide how much information is going to be default/basic printable.